Module-9/Homework9/PicProwlr/Views/CommonComponents/AnimatedProgressOwl.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  AnimatedProgressOwl.swift
+//  Created by Mark Renaud (2024).
+//
+
+
+import SwiftUI
+
+struct AnimatedProgressOwl: View {
+    /// The angle of rotation for the eyebrows in degrees.
+    @State private var eyebrowRotation: Double = -3.0
+    /// Value between 0.0 and 1.0 representing completion progress.
+    let progress: Double
+    let eyebrowAnimation = Animation.interactiveSpring.repeatForever(autoreverses: true)
+    
+    var body: some View {
+            ZStack(alignment: .center) {
+                ProgressOwl(progress: progress)
+                Image(.prowlrEybrows)
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .rotationEffect(.degrees(eyebrowRotation), anchor: .center)
+        }
+        .padding()
+        .onAppear {
+            withAnimation(eyebrowAnimation) {
+                eyebrowRotation *= -1.0
+            }
+        }
+    }
+}
+
+fileprivate struct PreviewHelper: View {
+    @State private var progress: Double = 0.0
+    var body: some View {
+        VStack {
+            AnimatedProgressOwl(progress: progress)
+            Button("Reset") {
+                progress = 0.0
+                withAnimation(.linear(duration: 10.0)) {
```

Comment:
Issue #20 - Fixed with 164eebd

Reply:
I forgot to mention the main motivation behind this. So for example, when having multiple smaller animations going on inside a bigger one, it's good to have them parametrised with a common multiplier. So when debugging, you can easily slow down just that animation just by changing one parameter! Or in case designer decides everything should be a little bit faster. Just a tip from my game development experience 😅